### DISCUSSION TITLE
Add a maparray.

### DISCUSSION THREAD

**Discussion by Orochikyocr** [ikemen-engine/Ikemen-GO#841] [Feature Request]
Let's say I need to trigger to activate a function.
trigger1 = var(5) = 10; single value
trigger1 = var(5) = [5,10]; range between 5 and 10 ,5,6,7,8,9,10

But what if I need only 5, 7, and 9 for whatever reason I would need such a thing, right now is like this.
trigger1  = var(5) = 5 || var(5) = 7 || var(5) = 9

I can think of two solutions.

trigger1 = var(5) = [5 || 7 || 9]

or an array

type = maparray
trigger1 = 1
map = "thevaluesneeded"
value = 5
value = 7 
value = 9

then 
trigger1 = var(5) = map(thevaluesneeded);

This should look way shorter, now imagine this with many values.

I'm working converting Stupa code to be used as a standard helper that any char could use,
since I needed to change how advframe sprites are being called, I needed to create a new
function that calls the same map, over and over. I just amazed at how nobody has think of adding arrays to cns or zss so far. 

![image](https://user-images.githubusercontent.com/42851314/82105733-0f2f0c80-96da-11ea-9198-163725429698.png)
This code could be way shorter with an array. Also, this is more likely to cause errors because a typo.

**Comment by Orochikyocr** [comment]
Another example of how array or || for trigger values will be useful
![image](https://user-images.githubusercontent.com/42851314/86289185-7f212580-bba8-11ea-8b8f-5fd57ce012a4.png)
I had this code to detect the enemy in three different states to set a nothitby.
This could be shortened to
"Ryo Sakazaki" && (enemynear,stateno = 300 || 650 || 440) 
Way simpler to read or even add more values. Thanks in advance.
PS: BTW sorry for the message in the discord, but I never say you dropped support for MAC, something got lost in translation, I meant you put on hold the support until you can find a solution for your dependency troubles.

**Reply by LasombraDemon** [reply]
> I just amazed at how nobody has think of adding arrays to cns or zss so far.

Actually Arrays are one of the things that I think could revolutionize the whole engine.

**Comment by Lazin3ss** [comment]
Checking for an array's content through an equal operator seems like a good idea to unhardcode command trigger behavior related to syntax, since it could simply return an array of all current commands being performed by the player, so syntax would be able to stay the same.

`command = "a"` => `{"recovery", "punch", "a", "b"} = "a"`

Arrays should be added as a new data type, as opposed to just some special behavior for maps. Maps could hold an array/table of values, in any case.
